### PR TITLE
Remove deprecated BCO export endpoint

### PR DESCRIPTION
**Pull request by martenson**
instead call `POST api/invocations/{id}/prepare_store_download` with payload: 
```
{
    model_store_format: bco.json
}
```

xref https://github.com/galaxyproject/galaxy/pull/14620#issuecomment-1279061251

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by nsoranzo** [comment]
@HadleyKing This may be of interest for you.

**Comment by davelopez** [comment]
I think we need to wait a bit more to drop this because the new endpoint requires Celery to be enabled in the server and usegalaxy.org is not using Celery yet AFAIK.

https://github.com/galaxyproject/galaxy/blob/9db2706e064c31da8c389405f1b1c95260963003/lib/galaxy/webapps/galaxy/services/invocations.py#L152

**Comment by davelopez** [comment]
@martenson, I think this can go now since ORG is running Celery :)

File: lib/galaxy_test/api/test_workflows.py [hunk 1]
```diff
@@ -3082,15 +3082,6 @@ def test_workflow_invocation_report_custom(self):
             assert "## Workflow Inputs" in markdown_content
             assert "## About This Report" in markdown_content
 
-    @skip_without_tool("cat1")
-    def test_export_invocation_bco(self):
```
Comment (davelopez) [inline lib/galaxy_test/api/test_workflows.py]: We should probably keep the test but update or replace `get_biocompute_object` function to use the short-term download API.

I guess calling `self.workflow_populator.download_invocation_to_store(invocation_id, extension="bco.json")` will do the trick.

https://github.com/galaxyproject/galaxy/blob/a5d66888c9ca37f821b8510ca4c4a5852ba2ff7d/lib/galaxy_test/base/populators.py#L1790

Comment (martenson) [reply lib/galaxy_test/api/test_workflows.py]: I saw the test below, but having an API test is surely a good idea, added a commit, thanks for the review!

https://github.com/galaxyproject/galaxy/blob/4e27e13b8e15e93e796cea074e7860c4a23f4ba2/test/integration/test_workflow_tasks.py#L66-L70

**Review by davelopez — APPROVED** [review]

**Pull request merged**